dm/ctl/simulator/event_filter.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simulator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/dm/dm/config"
+	"github.com/pingcap/dm/dm/ctl/common"
+	"github.com/pingcap/dm/dm/pb"
+
+	"github.com/pingcap/errors"
+	_ "github.com/pingcap/tidb/types/parser_driver" // for import parser driver
+	"github.com/spf13/cobra"
+)
+
+type eventFilterResult struct {
+	Result         bool   `json:"result"`
+	Msg            string `json:"msg"`
+	WillBeFiltered string `json:"will-be-filtered"`
+	FilterName     string `json:"filter-name,omitempty"`
+}
+
+// NewEventFilterCmd creates a EventFilter command
+func NewEventFilterCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "event-filter <config-file> <sql>",
+		Short: "check whether the given sql will be filtered by binlog-event-filter",
+		Run:   eventFilterFunc,
+	}
+	return cmd
+}
+
+func eventFilterFunc(cmd *cobra.Command, _ []string) {
+	if len(cmd.Flags().Args()) < 2 {
+		fmt.Println(cmd.Usage())
+		return
+	}
+	workers, err := common.GetWorkerArgs(cmd)
```

Comment:
Why get `worker`? It is not in the cmd.

Reply:
The binlog filter configs are applied on single worker. `worker` is defined in `rootCmd`.